### PR TITLE
Fix DebugEndpoint file leak

### LINKED ISSUE #3330: debug_endpoint does not close opened files
[`DebugEndpoint`](https://github.com/Tribler/tribler/blob/next/Tribler/Core/Modules/restapi/debug_endpoint.py) does not close its opened files:

https://github.com/Tribler/tribler/blob/486797b090c637e8c1b35059a8ce3962589b79b4/Tribler/Core/Modules/restapi/debug_endpoint.py#L347

https://github.com/Tribler/tribler/blob/486797b090c637e8c1b35059a8ce3962589b79b4/Tribler/Core/Modules/restapi/debug_endpoint.py#L400

https://github.com/Tribler/tribler/blob/486797b090c637e8c1b35059a8ce3962589b79b4/Tribler/Core/Modules/restapi/debug_endpoint.py#L403

Either `with` or `close` should be used. This is leaking file handles.

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #3330 

This PR closes the file handlers opened by the DebugEndpoint.

This PR also fixes the Windows unittest segfault.

### CODE REVIEW

**Comment by qstokkink** [comment]
@devos50 ready for review! Also, our Windows machine appears to have an old version of `meliae`. Nevertheless, this PR allows for a partial dump even using an old `meliae`.

File: Tribler/Core/Modules/restapi/debug_endpoint.py [hunk 1]
```diff
@@ -12,6 +14,12 @@
 import Tribler.Core.Utilities.json_util as json
 
 
+class MemoryDumpBuffer(StringIO):
```
Comment (devos50) [inline Tribler/Core/Modules/restapi/debug_endpoint.py]: Why do we need this `MemoryDumpBuffer` exactly?

Comment (qstokkink) [reply Tribler/Core/Modules/restapi/debug_endpoint.py]: `meliae` expects its file handle to support `write()`, `flush()` and `__call__()`, the `StringIO` class does not support the latter.

Comment (devos50) [reply Tribler/Core/Modules/restapi/debug_endpoint.py]: Could you add a comment in the code for that? 👍

Comment (qstokkink) [reply Tribler/Core/Modules/restapi/debug_endpoint.py]: Done 👍

**Comment by devos50** [comment]
@qstokkink thanks, I just went to EWI for a quick visit and fixed the Mac deployment tester. The fans of this machine were blowing at full speed so I assume some process was stuck internally. After restarting it, the fans went back to normal levels. I also checked the other mac whether there's a Tribler running in the background but I couldn't find it so there's something else that might be causing the failing GUI tests on Mac.

**Comment by qstokkink** [comment]
Hmm.. weird.

**Pull request merged**